### DISCUSSION TITLE
Info on hardware timers used for FreeRTOS on ESP32

### DISCUSSION THREAD

**Discussion by fdesvallees** [espressif/arduino-esp32#7223] [Q&A]
I am starting a project using a ESP-Wroom32 development board and FreeRTOS. I need to change the period of the timer that schedules FreeRTOS tasks,  etc.). How do I do this? Also, how do I find out which of the ESP32 hardware timers is used for this purpose?

Thanks!
François

**Comment by mrengineer7777** [comment]
Arduino-esp32 > esp32 > FreeRTOS.  At the Arduino level FreeRTOS should not be modified.  What are you trying to accomplish?

At the Arduino level you have Ticker:
https://github.com/espressif/arduino-esp32/tree/master/libraries/Ticker/src

At ESP32 level you can access hardware timers.
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/system/esp_timer.html

**Comment by fdesvallees** [comment]
Thank you, this is very helpful. I found that I can use another hardware timer for my purpose, and leave the system tick at its default value.
 
François
